paddle/memory/memory.h

```diff
@@ -29,10 +29,10 @@ void Free(Place, void*);
 template <typename Place>
 size_t Used(Place);
 
-template <typename T, /* must be POD types */
-          typename Place /* platform::GPUPlace or platform::CPUPlace */,
-          typename std::enable_if<std::is_pod<T>::value>::type* = nullptr>
+template <typename T, typename Place>
 class PODDeleter {
+  static_assert(std::is_pod<T>::value, "T must be POD");
```

Comment:
1. SFINAE is a versatile and powerful but potentially complicated.
2. `static_assert` makes more verbatim and hence clearer code.
3.  `static_assert` has the benefit of a clearer compiler error message